### DISCUSSION TITLE
How to skip over side buffers when navigating windows

### DISCUSSION THREAD

**Discussion by noahbald** [shortcuts/no-neck-pain.nvim#341] [Q&A] [answered]
I often use `<Ctrl-w>+k` or `<Ctrl-w>+l` to switch between neo-tree and my document when I need to. I don't particularly use the side buffers apart from centring my main buffer.
So I'm wondering if it's possible to skip over the side buffers when jumping through windows like this, so that I don't have to do `<Ctrl-w>+h/l` twice each time.

**Comment by shortcuts** [comment, marked as answer]
Hey! Thanks for using the plugin :)

What you are looking for is `autocmds.skipEnteringNoNeckPainBuffer`, take a look at https://github.com/shortcuts/no-neck-pain.nvim?tab=readme-ov-file#-configuration for more details :D

**Reply by noahbald** [reply]
Amazing, thanks pointing that out! I must have glossed over it 😅

**Comment by noahbald** [comment]
@shortcuts I seem to be running into a new issue now. It doesn't seem to skip over to neo-tree
 as if I've pressed `<Ctrl-w>+h` multiple times. It instead seems to snap back to my main buffer.
![Apr-04-2024 17-59-40](https://github.com/shortcuts/no-neck-pain.nvim/assets/36181524/49c5ed35-00fb-4f9c-9d43-d5f4d9c25503)

**Reply by shortcuts** [reply]
Hey @noahbald sorry for some reason I did not saw your reply! I fixed this issue in the next major version but will try to also provide a fix on the current version as next isn't ready to be stable yet

**Comment by janbuchar** [comment]
I'm going to piggyback on this one - any chance I could do the same, but with support for https://github.com/numToStr/Navigator.nvim? Currently, it's impossible to go to a tmux pane on the side with `autocmds.skipEnteringNoNeckPainBuffer` in place.

**Comment by noahbald** [comment]
fwiw I ended up getting a workaround for this, you can checkout this commit to my config [here](https://github.com/noahbald/LazyNvOAH/commit/2fcbc0d967c096feedf180e9ba5757034501ff50#diff-53d3e71e43de668dd337a95b2bffc2442eda80ff6ee548af5ab3118a065f36f3)
```lua
	{
		"nvim-neo-tree/neo-tree.nvim",
		opts = {
			event_handlers = {
				{
					event = "file_opened",
					handler = function()
						require("neo-tree.command").execute({ action = "close" })
					end,
				},
				{
					event = "neo_tree_buffer_enter",
					handler = function(arg)
						vim.opt.relativenumber = true
						vim.opt.number = true
					end,
				},
			},
			-- Add this line here
			open_files_do_not_replace_types = { "no-neck-pain" },
		},
		keys = {
			{ "<leader>e", false },
			{ "<leader>E", false },
		},
	},
```
